### PR TITLE
fix to disallow invalid characters in filenames

### LINKED ISSUE #143: indefinite progress bar when name typo in `zowe zos-ftp download data-set`
**Describe the bug**

Issue is being opened after an internal customer report. 

Customer accidentally entered the smiley face (ctrl-^B) into the command. Found that the command doesn't check for characters like emojis and therefore loops on the loading bar indefinitely. Had to exit with CRTL+C.
```
C:\Users\<USERINFO>\Test4z_test>zowe zos-ftp download data-set <FAKEDATASET> -f JOB52576.txt^B
Error: ENOENT: no such file or directory, open 'C:\Users\<USERINFO>\Test4z_test\JOB52576.txt☻'
Command Error:
ENOENT: no such file or directory, open 'C:\Users\<USERINFO>\Test4z_test\JOB52576.txt☻'
Imperative encountered an unexpected exception
Command entered: "zowe zos-ftp download data-set <FAKEDATASET> -f JOB52576.txt☻"
Use "zowe zos-ftp download --help" to view groups, commands, and options.
Error Details:
ENOENT: no such file or directory, open 'C:\Users\<USERINFO>\Test4z_test\JOB52576.txt☻'
 __________| 0%  o | Starting transfer...
```

**Expected and actual results**
There should be a thrown error indicating to user that their file name is invalid

**Describe your environment**
@zowe/zos-ftp-for-zowe-cli version  2.1.3

### PR DESCRIPTION
**Pull request by ATorrise**
**What It Does**
Provides a fix for [issue143](https://github.com/zowe/zowe-cli-ftp-plugin/issues/143) by throwing an error when using invalid filenames so the progress bar doesn't loop indefinitely from the unhandled case of emojis and other invalid characters in dataset names. Depends on [zowe-cli PR 1831](https://github.com/zowe/zowe-cli/pull/1831) - **unit tests will fail until this PR is merged in**.

**How to Test**
Tested with the following `launch.json` configurations

> ```
> 
>     {
>         "type": "node",
>         "request": "launch",
>         "name": "ftp download ds",
>         "program": "${workspaceFolder}/packages/cli/lib/main.js",
>         "console": "integratedTerminal",
>         "args": ["zos-ftp", "download", "data-set", <realTestingDataSetName>, "-f", "testing.txt^B", "--secure-ftp", "false"],
>         "outputCapture": "std",
>         "resolveSourceMapLocations": null
>     }
> ```
> Expected response: ```Data set downloaded successfully. testing.txt^B```
<br>

> ```
>         "args": ["zos-ftp", "download", "data-set", <realTestingDataSetName>, "-f", "testing.txt☻", "--secure-ftp", "false"],
> ```
> Expected response: ```Command Error: Invalid file name. Please check the file name for typos.```
<br>

> ```
>         "args": ["zos-ftp", "download", "data-set", <realTestingDataSetName>, "-f", "testing.txt", "--secure-ftp", "false"],
> ```
> Expected response: ```Data set downloaded successfully. testing.txt```

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/cli/Utilities.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+export class Utilities{
+    /**
+     * Check to ensure filename only contains valid characters
+     * @param {fileName} string - filename to be evaluated
+     * @returns {Promise<boolean>} - promise that resolves to true if filename contains valid characters
+     * @memberof Utilities
+     */
+    public static async isValidFileName(fileName: string): Promise<boolean> {
```
Comment (std4lqi) [inline src/cli/Utilities.ts:19]: It seems `async` is unnecessary here?

**Review by std4lqi — COMMENTED** [review]
Only one minor comments. Thanks!

File: src/cli/Utilities.ts (hunk 1 above)
Comment (ATorrise) [reply src/cli/Utilities.ts:19]: @std4lqi thank you!! copy pasted the method signature from a different file and didn't tailor anything but the method name 😅

File: src/cli/download/data-set/DataSet.Handler.ts [hunk 2]
```diff
@@ -14,33 +14,48 @@ import { FTPBaseHandler } from "../../../FTPBase.Handler";
...
 
 export default class DownloadDataSetHandler extends FTPBaseHandler {
     public async processFTP(params: IFTPHandlerParams): Promise<void> {
-
         const file = params.arguments.file == null ?
             ZosFilesUtils.getDirsFromDataSet(params.arguments.dataSet) :
             params.arguments.file;
+        try {
+            // Validate the destination file name before proceeding
+            if (!(await Utilities.isValidFileName(file))) {
```
Comment (std4lqi) [inline src/cli/download/data-set/DataSet.Handler.ts:27]: Good to remove `await`, I suppose.

**Review by std4lqi — APPROVED** [review]
A minor comment. Thanks, @ATorrise !

**Review by zFernand0 — CHANGES_REQUESTED** [review]
`@zowe/cli` version update required
waiting for this [workflow run](https://github.com/zowe/zowe-cli/actions/runs/6628035036/job/18017081565) to complete and for a new version out of [this PR](https://github.com/zowe/zowe-cli/pull/1831)

---

Now we only need to update the CLI version: `@zowe/cli@7.18.8`
- https://github.com/zowe/zowe-cli/pull/1831#issuecomment-1777980799

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**